numpy/lib/tests/test_loadtxt.py

```diff
@@ -1004,7 +1004,7 @@ def test_str_dtype_unit_discovery_with_converter():
     expected = np.array(
         ["spam-a-lot"] * 60000 + ["tis_but_a_scratch"], dtype="U17"
     )
-    conv = lambda s: s.strip("XXX")
+    conv = lambda s: s.removeprefix("XXX")
```

Comment:
Note that `s.removeprefix("XXX")` will remove a substring at the _beginning_ of the string only, while `s.strip("XXX")` will also remove characters at the _end_ of the string.

I think the _intent_ was to remove a substring at the beginning of the string. Please correct me if you don't agree.

Reply:
That is correct. In this particular constructed case, it doesn't really matter functionally as both will do the same on the constructed inputs, but `removeprefix("XXX")` is semantically closer to the intent.